### PR TITLE
BUG: fixes #1447 related to numpy.nan_to_num. 

### PR DESCRIPTION
**Pull request by janfb**
Add a test for this bug and fix the bug: if a finite integer number is passed to nan_to_num then the original number is returned in the format numpy.int64. Fix accomplished by moving the check for scalar above the first return statement. 
The test now ensures that no array is returned when nan_to_num is called with a finite scalar.

### CODE REVIEW

**Comment by seberg** [comment]
Sounds good, the test is broken on 32bit platforms/windows though.

You could also remove the whole isscalar stuff if you feel like doing a little maintenance, I cannot make out why it is there, I assume it is there because this used to use some boolean indexing (before copyto existed), which used to have problems with scalar boolean index into a 0-d array.

**Comment by charris** [comment]
Is #1447 correct? It looks unrelated at first sight.

**Comment by seberg** [comment]
Right, seems correct is gh-1477

**Comment by seberg** [comment]
Ah on first sight thought this was that it returned a 1-D array. Seems OK anyway, to always return a scalar. In which case the scalar logic could be replaced with indexing with `x[()]` as well and not removed completely as I first thought.

File: numpy/lib/tests/test_type_check.py [hunk 1]
```diff
@@ -278,10 +278,16 @@ def test_generic(self):
 
     def test_integer(self):
         vals = nan_to_num(1)
-        assert_all(vals == 1)
+        assert_all(type(vals) == np.int64)
```
Comment (charris) [inline numpy/lib/tests/test_type_check.py]: This fails on 32 bit linux and windows where the result is 32 bits. Need to take a closer look at this, the test is a bit odd...

**Comment by janfb** [comment]
There was a typo in the issue number. I forked again and will submit a new pull request on the correct issue number #1477. This pull request can be closed then.

**Comment by charris** [comment]
BTW, you can fix commit messages with `git rebase -i`  followed by a force push to origin.

**Pull request closed without merging**